fuel-core-interfaces/src/model/deposit_coin.rs

```diff
@@ -0,0 +1,20 @@
+use super::{BlockHeight, DaBlockHeight};
+use fuel_types::{Address, AssetId, Bytes32, Word};
+
+/// Probably going to be supperseaded with bridge message https://github.com/FuelLabs/fuel-core/issues/366
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone)]
+pub struct DepositCoin {
+    pub owner: Address,
+    pub amount: Word,
+    pub asset_id: AssetId,
+    pub nonce: Bytes32,
+    pub deposited_da_height: DaBlockHeight,
+    pub fuel_block_spend: Option<BlockHeight>,
+}
+
+impl DepositCoin {
+    pub fn id(&self) -> Bytes32 {
+        self.nonce
```

Comment:
why force copy? also, if the field and is public, why create a getter with a different name? maybe this will cause confusion

Reply:
It was some kind of placeholder as we were talking about hashing all fields in the hash but wasn't sure exactly what, but in the last iteration, we are removed `DepositCoin` and introduces bridge `Messages` so this part is going to be changed when details in contracts become known.